magic-nix-cache/src/gha.rs

```diff
@@ -224,7 +226,13 @@ fn path_info_to_nar_info(store: Arc<NixStore>, path_info: &ValidPathInfo, url: S
         references: path_info
             .references
             .iter()
-            .map(|r| r.file_name().unwrap().to_str().unwrap().to_owned())
+            .map(|r| {
+                r.file_name()
+                    .and_then(|n| n.to_str())
+                    .unwrap_or_else(|| panic!("failed to convert nar_info reference to string: {}",
+                        r.display()))
+                    .to_owned()
```

Comment:
this seems like somewhere we might want to surface out an error, maybe MNC should just warn + skip the path?

Maybe that's worse though, it'd be annoying to debug a case where 99% of a closure is uploaded but a ref is missing. Maybe Nix will give an okay message in that case.

Reply:
but not right now :)